sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -608,7 +608,8 @@ private[execution] final class LongToUnsafeRowMap(val mm: TaskMemoryManager, cap
   def optimize(): Unit = {
     val range = maxKey - minKey
```

Comment:
They're already `Long`s so `range` is too. The difference can overflow though. This should correctly handle the case.